hedera-node/src/main/java/com/hedera/services/contracts/sources/TxnAwareSoliditySigsVerifier.java

```diff
@@ -51,24 +47,23 @@
 	private final ActivationTest activationTest;
 	private final TransactionContext txnCtx;
 	private final BiPredicate<JKey, TransactionSignature> cryptoValidity;
-	private final Supplier<MerkleMap<EntityNum, MerkleToken>> tokens;
-	private final Supplier<MerkleMap<EntityNum, MerkleAccount>> accounts;
+	private WorldLedgers ledgers;
 
 	@Inject
 	public TxnAwareSoliditySigsVerifier(
 			final ActivationTest activationTest,
 			final TransactionContext txnCtx,
-			final Supplier<MerkleMap<EntityNum, MerkleToken>> tokens,
-			final Supplier<MerkleMap<EntityNum, MerkleAccount>> accounts,
 			final BiPredicate<JKey, TransactionSignature> cryptoValidity
 	) {
 		this.txnCtx = txnCtx;
-		this.tokens = tokens;
-		this.accounts = accounts;
 		this.activationTest = activationTest;
 		this.cryptoValidity = cryptoValidity;
 	}
 
+	public void setWorldLedgers(final WorldLedgers ledgers) {
```

Comment:
Since (1) component is now unusable without an appropriate `WorldLedgers` and (2) there is not a single instance of `WorldLedgers` that will always be appropriate; imo it makes the most sense to pass the needed `WorldLedgers` as an argument into each of the `has*()` methods defined in the `SoliditySigsVerifier` interface.

(This new argument will replace the `ContractAliases` argument, since a `WorldLedgers` instances has a [reference to the appropriate `ContractAliases`](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/store/contracts/WorldLedgers.java#L55).)